crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -125,6 +125,8 @@ impl Extension for QueryLimitsChecker {
             depth += 1;
             self.check_limits(cfg, num_nodes, depth, None)?;
             while level_len > 0 {
+                // Ok to unwrap since we checked for empty queue
```

Comment:
I can't quite figure it out, but this loop seems like it should just be a `while let Some(sel) = que.pop_front() { /* ... */ }` kind of loop, to avoid the need for an unwrap altogether.  I'll look harder when I go over the whole codebase while catching up.

Reply:
Not quite the same.
We're not trying to pop all from the queue, but only the N from the current level because the queue can contain the children also being pushed on line 135.

Reply:
Yeah that's what I meant by not being able to figure it out -- I'm thinking there's some rearrangement of the code that lets us avoid the `unwrap` by using a `while let Some(...) = ...pop() {` but it hasn't come to me yet.

Reply:
Another way is to use two queues; one for current level, other for children. Then `while let Some(...)` works well.